responsemanager/responsemanager_test.go

```diff
@@ -64,11 +77,20 @@ func TestIncomingQuery(t *testing.T) {
 	}
 	td.assertCompleteRequestWith(graphsync.RequestCompletedFull)
 
+	testSpan.End()
+	td.taskqueue.WaitForNoActiveTasks()
+	testutil.AssertDoesReceive(td.ctx, t, completedResponse, "request never completed")
+
 	// ensure request queued hook fires.
 	out := <-qhc
 	require.Equal(t, td.p, out.p)
 	require.Equal(t, out.request.ID(), td.requestID)
 	td.connManager.RefuteProtected(t, td.p)
+
+	tracing := collectTracing(t)
+	require.ElementsMatch(t, []string{
+		"TestIncomingQuery(0)->response(0)->executeTask(0)",
+	}, tracing.TracesToStrings())
```

Comment:
here we demonstrate the ability to insert an arbitrary parent to the `response` trace